apps/dashboard/src/components/workflow-list.tsx

```diff
@@ -170,7 +170,10 @@ export const WorkflowList = () => {
                               buildRoute(ROUTES.EDIT_WORKFLOW, {
                                 environmentId: currentEnvironment?._id ?? '',
                                 workflowId: workflow._id,
-                              })
+                              }),
+                              {
+                                state: { workflowName: workflow.name },
+                              }
```

Comment:
we will allow later updating of the workflow name in the drawer, so having this code won't work because the name won't be updated

Reply:
Lifting up the useFetchworkflow to the this component will also require to lift up other form states and all,
may we can have a global state or context of currentActivePage 🤔 wdyt?

Reply:
I'm not sure you need to lift up the form state, just the query call... the queries are created by the key, so it is fine to call `useQuery` multiple times, and the request will be triggered only once...  then we can update the cache and decide when to do it... because I'm not sure if it should be after every response from save (fyi we will have autosave)